contracts/Controller.sol

```diff
@@ -230,42 +235,63 @@ contract Controller is ReentrancyGuard, Ownable {
      * @dev For each action in the action Array, run the corresponding action
      * @param _actions An array of type Actions.ActionArgs[] which expresses which actions the user want to execute.
      */
-    function _runActions(Actions.ActionArgs[] memory _actions) internal {
+    function _runActions(Actions.ActionArgs[] memory _actions) internal returns (MarginAccount.Vault memory) {
+        MarginAccount.Vault memory vault;
+
+        uint256 prevActionVaultId;
+        bool isActionVaultStored;
+
         for (uint256 i = 0; i < _actions.length; i++) {
             Actions.ActionArgs memory action = _actions[i];
             Actions.ActionType actionType = action.actionType;
 
-            uint256 prevActionVaultId;
-
             if (actionType == Actions.ActionType.OpenVault) {
                 // check if this action is manipulating the same vault as all other actions, other than SettleVault
-                require(
-                    prevActionVaultId == 0 || action.vaultId == prevActionVaultId,
-                    "Controller: can not run actions on different vaults"
+                (prevActionVaultId, isActionVaultStored) = _checkActionsVaults(
+                    prevActionVaultId,
+                    action.vaultId,
+                    isActionVaultStored
                 );
-                prevActionVaultId = action.vaultId;
 
                 _openVault(Actions._parseOpenVaultArgs(action));
             }
         }
+
+        return vault;
```

Comment:
vault isn't initialized to anything though is it?

Reply:
because when user open a vault, it just increment vault counter in the mapping, doesn't modify any vault, in that case it will return an empty vault.

Reply:
ahh ok. for other actions it needs to be testsed differently then.